types/node/index.d.ts

```diff
@@ -977,6 +977,8 @@ declare namespace NodeJS {
 
         constructor(id: string, parent?: Module);
     }
+
+    type TypedArray = Uint8Array | Uint8ClampedArray | Uint16Array | Uint32Array | Int8Array | Int16Array | Int32Array | Float32Array | Float64Array;
```

Comment:
Could you explain why need to introduce new type definition instead of using `ArrayBufferView`, which covers the same types https://heycam.github.io/webidl/#common ?

Reply:
Hmm. Maybe something wrong, but in [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/ArrayBufferView) `ArrayBufferView` also contains `DataView`.

Reply:
From MDN docs point of view it looks nice but the definiton in lib.es5.d.ts is quite reduced compared to that:
```
interface ArrayBufferView {
    /**
      * The ArrayBuffer instance referenced by the array.
      */
    buffer: ArrayBufferLike;

    /**
      * The length in bytes of the array.
      */
    byteLength: number;

    /**
      * The offset in bytes of the array.
      */
    byteOffset: number;
}
```

See also https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26886#issuecomment-400853289 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26886#issuecomment-400989270

To my knowlege Node really checks on types in this so I think using a union of existing types matches better then the loose `ArrayBufferView` interface.

Reply:
Yeah. The actual one checks against the string name of the prototype rather than eg `ArrayBuffer.isView()`.